### PR TITLE
Feat/vmalert add max queue size

### PR DESCRIPTION
**Pull request by xbsura**
Im using vmalert-prod online, replacing our own version

find a littile little question, here:

1. some param name is not same with vm other utils, such as vmagent, we make it same
2. default maxQueueSize is not enough for us as we have thounds of alert rule, so make it as a param

thanks

### CODE REVIEW

File: app/vmalert/main.go [hunk 1]
```diff
@@ -37,17 +37,18 @@ absolute path to all .yaml files in root.`)
 	basicAuthUsername = flag.String("datasource.basicAuth.username", "", "Optional basic auth username for -datasource.url")
 	basicAuthPassword = flag.String("datasource.basicAuth.password", "", "Optional basic auth password for -datasource.url")
 
-	remoteWriteURL = flag.String("remotewrite.url", "", "Optional URL to Victoria Metrics or VMInsert where to persist alerts state"+
+	remoteWriteURL = flag.String("remoteWrite.url", "", "Optional URL to Victoria Metrics or VMInsert where to persist alerts state"+
```
Comment (hagen1778) [inline app/vmalert/main.go:40]: please update README according to flag rename

File: app/vmalert/main.go [hunk 2]
```diff
@@ -37,17 +37,18 @@ absolute path to all .yaml files in root.`)
...
 		" in form of timeseries. E.g. http://127.0.0.1:8428")
-	remoteWriteUsername = flag.String("remotewrite.basicAuth.username", "", "Optional basic auth username for -remotewrite.url")
-	remoteWritePassword = flag.String("remotewrite.basicAuth.password", "", "Optional basic auth password for -remotewrite.url")
+	remoteWriteUsername     = flag.String("remoteWrite.basicAuth.username", "", "Optional basic auth username for -remotewrite.url")
+	remoteWritePassword     = flag.String("remoteWrite.basicAuth.password", "", "Optional basic auth password for -remotewrite.url")
+	remoteWriteMaxQueueSize = flag.Int("remoteWrite.maxQueueSize", 100, "Optional remote write maxQueueSize, default 100, you need to make it bigger if push print err: `failed to push timeseries to remotewrite: failed to push timeseries - queue is full`")
```
Comment (hagen1778) [inline app/vmalert/main.go:44]: I'd suggest following:
* description - `Defines the max number of pending datapoints to remote write endpoint.`
* default value - 10000; this should be enough for 99% of cases and doesn't look expensive in terms of memory
* drop `you need to make it bigger if push print err: `failed to push timeseries to remotewrite: failed to push timeseries - queue is full` hint from description and add recommendation to increase `remoteWrite.maxQueueSize` value into error message itself.

**Comment by hagen1778** [comment]
@tenmozes these are breaking changes to the flag names. I'd prefer to avoid this but the change will make it consistent with `vmagent` flag names. So it would be better to do this now. Please, update the helm-chart accordingly after this being merged.

File: app/vmalert/main.go (hunk 2 above)
Comment (xbsura) [reply app/vmalert/main.go:44]: ok

**Comment by hagen1778** [comment]
Thanks!

**Pull request merged**